### PR TITLE
Offline detector shouldn`t be rendered in SSR

### LINKED ISSUE #811: Offline detector shouldn`t be rendered in SSR

### PR DESCRIPTION
**Pull request by Katrin-kudryash**
## PR includes

- [x] Bug Fix

## Related issues

Resolve #811

### CODE REVIEW

File: src/hooks/useOfflineDetector.tsx [hunk 1]
```diff
@@ -11,7 +11,12 @@ export const useOfflineDetector = ({
     pollingDelay = 5000,
     remoteServerUrl = '/',
 }: UseOfflineDetectorProps) => {
-    const [globalOnlineStatus, setGlobalOnlineStatus] = useState(navigator.onLine);
+    const [globalOnlineStatus, setGlobalOnlineStatus] = useState(() => {
+        if (typeof window === 'undefined') {
+            return;
```
Comment (tsumo) [inline src/hooks/useOfflineDetector.tsx:16]: return true

File: src/hooks/useOfflineDetector.tsx [hunk 2]
```diff
@@ -11,7 +11,12 @@ export const useOfflineDetector = ({
...
+        }
+        return true;
```
Comment (tsumo) [inline src/hooks/useOfflineDetector.tsx:18]: return navigator.online

File: src/hooks/useOfflineDetector.tsx [hunk 3]
```diff
@@ -36,11 +41,11 @@ export const useOfflineDetector = ({
                 fetch(remoteServerUrl, { method: 'HEAD' })
                     .then((res) => {
                         setRemoteServerStatus(res.status < 400);
-                        setStatus?.([globalOnlineStatus, true]);
+                        setStatus?.([!!globalOnlineStatus, true]);
```
Comment (tsumo) [inline src/hooks/useOfflineDetector.tsx:44]: unnecessary

File: src/hooks/useOfflineDetector.tsx [hunk 4]
```diff
@@ -36,11 +41,11 @@ export const useOfflineDetector = ({
...
                     })
                     .catch(() => {
                         setRemoteServerStatus(false);
-                        setStatus?.([globalOnlineStatus, false]);
+                        setStatus?.([!!globalOnlineStatus, false]);
```
Comment (tsumo) [inline src/hooks/useOfflineDetector.tsx:48]: same

**Review by tsumo — CHANGES_REQUESTED** [review]

**Review by tsumo — APPROVED** [review]

**Review by 9teen90nine — APPROVED** [review]

**Pull request merged**